scripts/full_check.sh

```diff
@@ -0,0 +1,110 @@
+#! /bin/bash
+
+# Copyright (C) 2024 Intel Corporation
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+# http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# 
+# SPDX-License-Identifier: Apache-2.0
+
+
+# this is for developer local check only.
+# not used by github action
+
+check_tool() {
+  which $1 &> /dev/null || (echo "$1 not found!" && exit 1)
+}
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --format)
+      CHECK_FORMAT=1
+      shift
+      ;;
+    --tidy)
+      CHECK_TIDY=1
+      shift
+      ;;
+    --license)
+      CHECK_LICENSE=1
+      shift
+      ;;
+    *) # Handle other unknown options
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+# set PR_REF if your pr target is not main
+: ${PR_REF:=main}
+
+
+PROJECT_ROOT=$(pwd)/$(dirname "${BASH_SOURCE[0]}")/../
```

Comment:
This will not work if the script is run with the full path, e.g. /a/b/c/full_check.sh you will get /cur/dir//a/b/c/../.
I'd avoid using bash syntax, when you could do the same with POSIX, especially when it's more simple.
```suggestion
PROJECT_ROOT="$(cd "$(dirname "$0")" && pwd)"
```